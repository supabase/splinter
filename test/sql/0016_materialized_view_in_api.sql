begin;
  set local search_path = '';
  set local pgrst.db_schemas = 'public';

  -- 0 issues
  select * from lint."0016_materialized_view_in_api";

  create materialized view public.my_view as select 1;

  -- 1 issue
  select * from lint."0016_materialized_view_in_api";

  -- Resolve the issue with permissions
  revoke select on public.my_view from anon, authenticated, public;

  -- 0 issues
  select * from lint."0016_materialized_view_in_api";


rollback;
