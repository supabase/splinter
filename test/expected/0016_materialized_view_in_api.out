begin;
  set local search_path = '';
  set local pgrst.db_schemas = 'public';
  -- 0 issues
  select * from lint."0016_materialized_view_in_api";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create materialized view public.my_view as select 1;
  -- 1 issue
  select * from lint."0016_materialized_view_in_api";
           name           |          title           | level |  facing  | categories |                              description                              |                                      detail                                       |                                         remediation                                          |                               metadata                               |                cache_key                
--------------------------+--------------------------+-------+----------+------------+-----------------------------------------------------------------------+-----------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------+----------------------------------------------------------------------+-----------------------------------------
 materialized_view_in_api | Materialized View in API | WARN  | EXTERNAL | {SECURITY} | Detects materialized views that are potentially accessible over APIs. | Materialized view \`public.my_view\` is selectable by anon or authenticated roles | https://supabase.com/docs/guides/database/database-linter?lint=0016_materialized_view_in_api | {"name": "my_view", "type": "materialized view", "schema": "public"} | materialized_view_in_api_public_my_view
(1 row)

  -- Resolve the issue with permissions
  revoke select on public.my_view from anon, authenticated, public;
  -- 0 issues
  select * from lint."0016_materialized_view_in_api";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
